### PR TITLE
Defer dashboard deployment until "minikube dashboard" is executed

### PR DESCRIPTION
**Pull request by tstromberg**
This simplifies cluster startup, and saves about 20MB resident memory (1% of our default VM size). More importantly, it avoids starting unused services, downloading unnecessary files, and gumming the logs up with possibly unimportant error messages relating to them.

This PR doesn't change the behavior substantially, as currently minikube starts the dashboard pod, but only blocks until it's healthy when the "dashboard" command is executed. This PR simply defers the startup as well as the health check.

### CODE REVIEW

File: cmd/minikube/cmd/dashboard.go [hunk 1]
```diff
@@ -65,19 +65,30 @@ var dashboardCmd = &cobra.Command{
 		}
 		cluster.EnsureMinikubeRunningOrExit(api, 1)
 
+		fmt.Fprintln(os.Stderr, "Enabling dashboard ...")
+		// Enable the dashboard add-on
+		err = configcmd.Set("dashboard", "true")
```
Comment (balopat) [inline cmd/minikube/cmd/dashboard.go]: is this idempontent? If yes, then we don't need "Enabling dashboard ..." the second time (I think addons enable is idempotent) - would be nice to check for enablement.

File: cmd/minikube/cmd/dashboard.go [hunk 2]
```diff
@@ -65,19 +65,30 @@ var dashboardCmd = &cobra.Command{
...
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Unable to enable dashboard: %v\n", err)
+			os.Exit(1)
+		}
+
 		ns := "kube-system"
 		svc := "kubernetes-dashboard"
-		if err = util.RetryAfter(30, func() error { return service.CheckService(ns, svc) }, 1*time.Second); err != nil {
+		fmt.Fprintln(os.Stderr, "Verifying dashboard health ...")
+		if err = util.RetryAfter(180, func() error { return service.CheckService(ns, svc) }, 1*time.Second); err != nil {
```
Comment (balopat) [inline cmd/minikube/cmd/dashboard.go]: It would be interesting to keep a "watch" on the dashboard pod + on a separate line the result of the service status? But this idea might be better off in a separate PR. It's just 3 minutes of potential waiting is definitely long. Out of those 1 minute is almost guaranteed to be just the addon-manager kicking in...

File: cmd/minikube/cmd/dashboard.go [hunk 3]
```diff
@@ -65,19 +65,30 @@ var dashboardCmd = &cobra.Command{
...
 			fmt.Fprintf(os.Stderr, "%s:%s is not running: %v\n", ns, svc, err)
 			os.Exit(1)
 		}
 
+		fmt.Fprintln(os.Stderr, "Launching proxy ...")
```
Comment (balopat) [inline cmd/minikube/cmd/dashboard.go]:
```suggestion
		fmt.Fprintln(os.Stderr, "Launching dashboard proxy ...")
```

File: cmd/minikube/cmd/dashboard.go [hunk 4]
```diff
@@ -65,19 +65,30 @@ var dashboardCmd = &cobra.Command{
...
 		p, hostPort, err := kubectlProxy()
 		if err != nil {
 			glog.Fatalf("kubectl proxy: %v", err)
 		}
 		url := dashboardURL(hostPort, ns, svc)
 
+		fmt.Fprintln(os.Stderr, "Verifying proxy health ...")
```
Comment (balopat) [inline cmd/minikube/cmd/dashboard.go]:
```suggestion
		fmt.Fprintln(os.Stderr, "Verifying dashboard proxy health ...")
```

File: cmd/minikube/cmd/dashboard.go (hunk 2 above)
Comment (tstromberg) [reply cmd/minikube/cmd/dashboard.go]: A good idea, but due to the complexity l will leave this for another PR. This step generally only takes a second or two, but to avoid flakes in places with poor connectivity, I wanted to make sure to wait an extended period for the pod to come up.

**Comment by afbjorklund** [comment]
A lot of people loved having features like the cache and the dashboard available by default, but I guess if it clearly documented how to turn them on again they can still do it on the command-line (or auto-enable). Guess it was features like heapster that made it too heavy , don't recall the dashboard itself being too bad ? Especially not if comparing to the boot time and resource consumption of the VM...

**Comment by tstromberg** [comment]
@afbjorklund - Thanks for raising the point that the PR title/description was not well crafted. This doesn't effectively change the behavior to users: the dashboard command will still work just as it has. It simply delays the dashboard pod startup until the user first wants to use it.

**Comment by tstromberg** [comment]
@minikube-bot OK to test
